src/m2n/PointToPointCommunication.cpp

```diff
@@ -300,7 +300,7 @@ void PointToPointCommunication::acceptConnection(std::string const &acceptorName
     // Establish connection between participants' master processes.
     auto c = _communicationFactory->newCommunication();
 
-    c->acceptConnection(acceptorName, requesterName, utils::MasterSlave::getRank());
+    c->acceptConnection(acceptorName, requesterName, _mesh->getName(), utils::MasterSlave::getRank());
```

Comment:
Not completely sure, but might be dangerous to use the mesh name here as it could interfere with the slaves communication below. It's a bit ugly here as `c` is again a (temporary) master-master communication. What about sth like `"TMP-MASTERCOM-" +_mesh->getName()`?

Reply:
Got it. I implemented your suggestion.